docs/proposals/storage_acls.md

```diff
@@ -0,0 +1,173 @@
+     * Storage Classes
+4. How much storage the user may consume: 
+      * Limits: enumerates resource constraints per project 
+       * Size of request
+       * Enforcement by Quota that calculates usage
+       * Number of requests/claims
+
+This PR will focus on providing a means  of Access Control Lists (ACLs) based on leveraging and extending Resource Quotas:
+```
+apiVersion: v1
+kind: ResourceQuota
+namespace: $the_users_namespace
+metadata:
+  name: storage-resource
+spec:
+  hard:
+    storageClass:
+      Gold:
+        requests.storage: 1
+        limits.storage: 2Gi
+      Silver:
+        requests.storage: 2
+        limits.storage: 2Gi
+      nil:
+        requests.storage: 2
+        limits.storage: 4Gi
+
+apiVersion: v1
+kind: VolumeSecurityPolicy
+namespace: $the_users_namespace
+metadata:
+  name: my-vsp
+spec:
+  StorageClass:
+    - Gold
+    - Silver
+    - nil
+
+```
+
+### Use Cases for ‘USE’:
+
+1. Use Case: Administrator wishes to create non-default provisioner and restrict it’s use to a specific namespace
+   * Default behavior: Disabled for ALL users
+2. Use Case: Administrator wishes to create DEFAULT provisioner 
+   * Default behavior: If SINGLE cloud provider is defined, default class will be created with these credentials
+   * Default behavior: Enabled for ALL users
+   * Controls: # of claims restricted by limits/quotas
+   * Special considerations:
+     * What do we do when there is more than one cloud provider defined?
+     * What do we do when there is NO cloud provider defined
+3. Use Case: Administrator wishes to delete provisioner
+   * Error message should be returned to user that provisioner unavailable. 
+   * Pending claims should be rejected with same message
+   * What happens if Administrator deleted default?
+   * Should default to storage class ‘nil’
+4. Use Case: Administrator wishes to change a parameter of the provisioner with minimal disruption
+   * Possible parameters: credentials, allowed users
+5. Use Case: Administrator wishes to limit the requests of a storage class based on namespace
+6. Use Case: Administrator wishes to restrict storage allocation for a given storage class based on namespace 
+7. Use Case: Administrator wishes to restrict requests (claims) for a stoage based on namespace
+8. Use Case: User wishes to remove claim to storage triggering a ‘detach’ 
+9. Use Case: Administrator updates GID in SCC for block storage to new GID, access to the storage should continue to work with the current pod spec 
+10. Use Case: PV having GID that isn’t part of the range defined in the SCC and binding to a PV but not usable by the pod (Important customer use case this doesn’t address today)
+
+### Special considerations for ‘USE’
+* Augment Quotas to include Storage with sub-classification for Storage Classes
+* Update default SCC to include binary value to enable/disable Storage Selectors
+* Augment Limits to include storage with sub-limits per Storage Class
```

Comment:
we will need to use the same resource name from https://github.com/kubernetes/kubernetes/pull/34554 in the limit range